src/ParallelDataCollector.cpp

```diff
@@ -559,6 +559,22 @@ namespace splash
                 localSize, globalOffset, sizeRead, ndims, buf);
     }
 
+     CollectionType* ParallelDataCollector::readMeta(int32_t id,
+             const char* name,
+             const Dimensions dstBuffer,
+             const Dimensions dstOffset,
+             Dimensions& sizeRead) throw (DCException)
+     {
+         // TODO: Check what this line does (FST_MERGING is not defined!)
+//       if (fileStatus != FST_READING && fileStatus != FST_WRITING && fileStatus != FST_MERGING)
```

Comment:
should basically check if the PDC object is valid and a file is opened.

You can copy that from `ParallelDataCollector::read`